### PR TITLE
feat: port media/formats/webm to CMake

### PR DESCRIPTION
**Pull request by cadubentzen**
Port media/formats/webm to CMake.

- [x] Compiles
- [x] Tests compile and pass

Depends on #1143 
Related #1047

### CODE REVIEW

**Comment by joeyparrish** [comment]
Will review in detail after #1143 lands.  Thanks!

**Comment by joeyparrish** [comment]
`formats_webm.lib(mkv_writer.obj) : error LNK2019: unresolved external symbol "public: static __int64 __cdecl shaka::File::CopyFileA(class shaka::File *,class shaka::File *,__int64)" (?CopyFileA@File@shaka@@SA_JPEAV12@0_J@Z) referenced in function "public: __int64 __cdecl shaka::media::MkvWriter::WriteFromFile(class shaka::File *,__int64)" (?WriteFromFile@MkvWriter@media@shaka@@QEAA_JPEAVFile@3@_J@Z) [D:\a\shaka-packager\shaka-packager\build\packager\media\formats\webm\webm_unittest.vcxproj]`

I'm guessing this is a Windows macro gone wrong.  `CopyFileA` smacks of the days of Microsoft API variants for ASCII  (A) and wide characters (W).

There's a macro in file/file.cc that deals with this, but it may need to be moved to the file.h header instead.

**Comment by cadubentzen** [comment]
@joeyparrish is finally green, please take a look :slightly_smiling_face:

**Comment by joeyparrish** [comment]
Will do.  Thanks!

File: packager/media/formats/webm/webm_cluster_parser.cc [hunk 1]
```diff
@@ -259,7 +259,7 @@ bool WebMClusterParser::OnBinary(int id, const uint8_t* data, int size) {
       return true;
 
     case kWebMIdBlockAdditional: {
-      uint64_t block_add_id = base::HostToNet64(block_add_id_);
+      uint64_t block_add_id = absl::big_endian::FromHost16(block_add_id_);
```
Comment (joeyparrish) [inline packager/media/formats/webm/webm_cluster_parser.cc:262]: Looks like this 16 should have been 64.

**Review by joeyparrish — CHANGES_REQUESTED** [review]
Looks like maybe one mistake.  The rest looks good.

File: packager/media/formats/webm/webm_cluster_parser.cc (hunk 1 above)
Comment (joeyparrish) [reply packager/media/formats/webm/webm_cluster_parser.cc]: Fixed.  If the tests pass, I'll merge it.

**Review by joeyparrish — APPROVED** [review]

File: packager/media/formats/webm/webm_cluster_parser.cc (hunk 1 above)
Comment (cadubentzen) [reply packager/media/formats/webm/webm_cluster_parser.cc]: Thanks for fixing it!

**Pull request merged**